### PR TITLE
refine pttdb

### PR DESCRIPTION
**Pull request by chhsiao1981**
This PR is related to #261

### CODE REVIEW

**Comment by chhsiao1981** [comment]
unable to pass the e2e-test.
close for now.

**Comment by chhsiao1981** [comment]
actually the PR can pass the e2e-test.

Found that add-friend failed when the user is with image,
which is not the scope of the PR.

File: pttdb/ldb_database_test.go [hunk 1]
```diff
@@ -1387,3 +1264,181 @@ func TestLDBBatch_TryPutAll(t *testing.T) {
 
 	// teardown test
 }
+
+func TestLDBBatch_TryPutAll2(t *testing.T) {
```
Comment (poga) [inline pttdb/ldb_database_test.go]: why is this test called `TryPutAll2` if `TryPutAll` is already removed?

File: pttdb/errors.go [hunk 2]
```diff
@@ -25,4 +25,5 @@ var (
 	ErrInvalidLock     = errors.New("invalid db lock")
 	ErrBusy            = errors.New("db busy")
 	ErrInvalidKeys     = errors.New("invalid db keys")
+	ErrInvalidIndex    = errors.New("invalid db index")
```
Comment (poga) [inline pttdb/errors.go]: What's the difference between db key and db index? can you add comments on what they are?

Comment (chhsiao1981) [reply pttdb/errors.go]: ErrInvalidIndex is the error when Index cannot be Unmarshalled.
ErrInvalidKeys is the error when the 2nd-indexing-key is invalid in LDBBatch.GetByIdxKey and GetKeyByIdxKey.


In LDBBatch.TryPutAll and LDBBatch.ForcePutAll:

idxKey, idx, and kvs intends to implement 2nd-indexing in leveldb.

Given the id-key and the data with updateTS,
idxKey is the id-key, key is the key-with-updateTS, value is the corresponding data.
Besides, we also have the 2nd-idx-keys to index by the 2nd-idx-keys.
The value of the 2nd-idx-keys are key.

idx includes the key (in the 0th of idx.Keys) and 2nd-idx-keys
kvs are the corresponding key-vals based on the order of idx.

File: pttdb/ldb_database_test.go (hunk 1 above)
Comment (chhsiao1981) [reply pttdb/ldb_database_test.go]: TryPutAll is actually not removed.

The removed part is the duplicated code.

**Pull request merged**